contributingGuides/TS_CHEATSHEET.md

```diff
@@ -213,3 +216,21 @@
   const array1 = ["hello", 1]; // type: (string | number)[]
   const array2 = ["goodbye", 2]; // type: readonly ["goodbye", 2]
   ```
+
+<a name="higher-order-components"></a><a name="1.9"></a>
+
+- [1.9](#higher-order-components) **Higher Order Components**
+
+  Typing HOCs is hard. Refer to [this articles](https://medium.com/@hayata.suenaga/ts-higher-order-components-30c38dd19ae8) for detailed guideline on typing HOCs for different usages of HOCs.
+
+<a name="function-overloading"></a><a name="1.10"></a>
+
+- [1.10](#function-overloading) **Function Overloading**
+
+  Use [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) to provide more type information for functions. For the following types of functions, function overloading can be beneficial.
+
+  - The return type depends on the input type
+  - When function accept different number of parameters
+  - There are type dependencies between parameters
+
+  Refer to [this guide](https://medium.com/@hayata.suenaga/when-to-use-function-overloads-acc48f7e3142) to learn how to use functional overloads for each situation.
```

Comment:
These two sections are nice addition to the guidelines, great job @hayata-suenaga! 🚀

Reply:
Awesome @hayata-suenaga 🎉